capycli/common/capycli_bom_support.py

```diff
@@ -355,9 +356,14 @@ def get_ext_ref(comp: Component, type: ExternalReferenceType, comment: str) -> O
     @staticmethod
     def set_ext_ref(comp: Component, type: ExternalReferenceType, comment: str, value: str,
                     hash_algo: str = "", hash: str = "") -> None:
+
+        if re.search(XsUri._INVALID_URI_REGEX, str(value)):
+          cleaned_uri = re.sub(XsUri._INVALID_URI_REGEX, ':', str(value))
```

Comment:
If I get this right, `XsUri._INVALID_URI_REGEX` mainly checks for invalid percent encoding in a URL, so I'm not sure how this would apply to #70. Also, I'm a bit confused why you replace invalid URL parts with a ":" here?

Reply:
Hi @gernot-h Try to take a look at the original CycloneDX code of XsUri. They do this check and then throw an exception.

Reply:
I did, and I don't yet see how replacing invalid percent encoding by ":" can help here, but I will try to reproduce #70 as you suggested and then debug it on myself. :-)

Reply:
Updated fix available in #72.